### ISSUE TITLE
Enhancement: Better error message if parserOptions.project is set to an invalid TSConfig

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10511]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

It's not unheard of for folks to set [`parserOptions.project`](https://typescript-eslint.io/packages/parser#project) to a different config file than a TSConfig. For example, someone could easily accidentally do:

```ts
parserOptions: {
  project: 'eslint.config.js',
}
```

...at which point their config would give a lovely spew like:

```plaintext
/Users/josh/repos/repros/index.ts
  0:0  error  Parsing error: eslint.config.js(1,1): error TS1327: String literal with double quotes expected.
eslint.config.js(1,8): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
eslint.config.js(1,15): error TS1136: Property assignment expected.
eslint.config.js(1,32): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
eslint.config.js(2,1): error TS1327: String literal with double quotes expected.
eslint.config.js(2,8): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
eslint.config.js(2,17): error TS1136: Property assignment expected.
eslint.config.js(2,41): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
eslint.config.js(4,1): error TS1327: String literal with double quotes expected.
eslint.config.js(4,7): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
eslint.config.js(4,8): error TS1327: String literal with double quotes expected.
eslint.config.js(4,16): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal
```

Yuck.

Proposal: if `parserOptions.project` is set to something that's not a valid TSConfig, how about the tooling give a specialized error message?

Vaguely:

```plaintext
/Users/josh/repos/repros/index.ts
  0:0  error  Parsing error: Unable to load parserOptions.project: "eslint.config.js" as a TSConfig. Is this a valid JSON(C) TSConfig file?
See https://typescript-eslint.io/getting-started/typed-linting for information on typed linting.
```

### Additional Info

Ideated out of this Discord thread: https://discord.com/channels/1026804805894672454/1317907420403732571

💖

**Comment by kirkwaiblinger** [comment]
+1

**Comment by bradzacher** [comment]
what constitutes a 'valid tsconfig' though...?
there's no filename restriction -- it just has to be a file with JSON contents really.
how deep should the  validation go?

**Comment by JoshuaKGoldberg** [comment]
I suppose we could just go off of parsing. If an error is thrown, it's invalid.

**Comment by ronami** [comment]
> what constitutes a 'valid tsconfig' though...? there's no filename restriction -- it just has to be a file with JSON contents really. how deep should the validation go?

+1 that we can't rely on the filename as it has no restrictions, but the contents of the file do (it has to be somewhat of a JSON file).

I think that, often, when developers use the wrong file as their `tsconfig.json`, it won't be a valid JSON file (as in the Discord thread). One option I can think of, is to add some context on the potential solution [when this does happen](https://github.com/typescript-eslint/typescript-eslint/blob/d24a82854d06089cbd2a8801f2982fd4781f3701/packages/typescript-estree/src/create-program/getParsedConfigFile.ts#L46), so this can look something along:

```
/test-project/src/index.ts
  0:0  error  Parsing error: Unable to parse the specified 'tsconfig' file. Ensure it's correct and has valid syntax.

../../foo.mjs(1,1): error TS1136: Property assignment expected.
../../foo.mjs(1,8): error TS1327: String literal with double quotes expected.
../../foo.mjs(1,8): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal

✖ 1 problem (1 error, 0 warnings)
```

Currently, the raw diagnostics are logged without much context:

```
/test-project/src/index.ts
  0:0  error  Parsing error: ../../foo.mjs(1,1): error TS1136: Property assignment expected.
../../foo.mjs(1,8): error TS1327: String literal with double quotes expected.
../../foo.mjs(1,8): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal

✖ 1 problem (1 error, 0 warnings)
```

The error is somewhat confusing because it may seem the linted file failed to parse, whereas the issue is actually with the provided tsconfig file. Formatting is somewhat off too for the first line.
